diesel/src/mysql/connection/raw.rs

```diff
@@ -142,6 +138,35 @@ impl RawConnection {
             ))
         }
     }
+
+    fn flush_pending_results(&self) -> QueryResult<()> {
+        self.consume_current_result()?;
+        while self.next_result()? {
+            self.consume_current_result()?;
+        }
+        self.consume_current_result()?;
+        Ok(())
+    }
+
+    fn consume_current_result(&self) -> QueryResult<()> {
+        unsafe {
+            let res = ffi::mysql_store_result(self.0);
+            if !res.is_null() {
+                ffi::mysql_free_result(res);
+            }
+            self.did_an_error_occur()
```

Comment:
This is not actually unsafe to call, is it? (I'd keep the unsafe section as short as possible)